### PR TITLE
New PCB Canary60RGB from CanaryTeam

### PR DESCRIPTION
**Pull request by AnthonyNguyen168**
## Description
New PCB CANARY60RGB from CanaryTeam
It is a hotswappable layout 60 PCB, which support VIA and RGB per key.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation (just documentation for my keyboard)
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/canary/canary60rgb/v1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (zvecr) [inline keyboards/canary/canary60rgb/v1/rules.mk:11]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also,

- All `*.c` and `*.h` files need GPL compliant headers
-  All `*.jpg` files need to be removed

File: keyboards/canary/canary60rgb/canary60rgb.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
+#include "canary60rgb.h"
```
Comment (drashna) [inline keyboards/canary/canary60rgb/canary60rgb.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/canary/canary60rgb/canary60rgb.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/canary/canary60rgb/canary60rgb.c:104]: This is no longer needed. 
```suggestion
```

File: keyboards/canary/canary60rgb/canary60rgb.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#pragma once
```
Comment (drashna) [inline keyboards/canary/canary60rgb/canary60rgb.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/canary/canary60rgb/config.h [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (drashna) [inline keyboards/canary/canary60rgb/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/canary/canary60rgb/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/canary/canary60rgb/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/canary/canary60rgb/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,39 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/canary/canary60rgb/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/canary/canary60rgb/v1/config.h [hunk 8]
```diff
@@ -0,0 +1,73 @@
+#pragma once
```
Comment (drashna) [inline keyboards/canary/canary60rgb/v1/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/canary/canary60rgb/v1/config.h [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+#define MATRIX_ROW_PINS { F5, F4, F1, B3, B2 }
+#define MATRIX_COL_PINS { C7, F7, F6, F0, B0, B1, B4, D7, D6, D4, D5, D3, D2, B7 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN B6
+#ifdef RGB_DI_PIN
+#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/canary/canary60rgb/v1/config.h:33]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/canary/canary60rgb/v1/rules.mk [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```
Comment (drashna) [inline keyboards/canary/canary60rgb/v1/rules.mk:17]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/canary/canary60rgb/v1/rules.mk [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no              # Audio output on port C6
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
+RGB_MATRIX_DRIVER = IS31FL3733
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
```
Comment (drashna) [inline keyboards/canary/canary60rgb/v1/rules.mk:32]:
```suggestion
BLUETOOTH_ENABLE = no          # Enable Bluetooth
AUDIO_ENABLE = no              # Audio output
RGB_MATRIX_ENABLE = yes        # Use RGB matrix
RGB_MATRIX_DRIVER = IS31FL3733
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/canary/canary60rgb/canary60rgb.c (hunk 2 above)
Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/canary60rgb.c]: your comment is resolved

File: keyboards/canary/canary60rgb/canary60rgb.c (hunk 3 above)
Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/canary60rgb.c]: This is resolved.

File: keyboards/canary/canary60rgb/canary60rgb.h (hunk 4 above)
Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/canary60rgb.h]: This is resolved.

**Comment by AnthonyNguyen168** [comment]
Dear maintainers,
I resolve all your comments with care.
Can you check this pull request again?
Thanks

File: keyboards/canary/canary60rgb/v1/config.h [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+#    define DISABLE_RGB_MATRIX_SPLASH
+#    define DISABLE_RGB_MATRIX_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_SOLID_SPLASH
+#    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_DIGITAL_RAIN
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_ADDR_2 0b1010000 // this is here for compliancy reasons.
+#    define DRIVER_COUNT 2
+#    define DRIVER_1_LED_TOTAL 63
+#    define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
```
Comment (fauxpark) [inline keyboards/canary/canary60rgb/v1/config.h:96]:
```suggestion
#    define DRIVER_COUNT 1
#    define DRIVER_LED_TOTAL 63
```

File: keyboards/canary/canary60rgb/info.json [hunk 13]
```diff
@@ -0,0 +1,80 @@
+{
+    "keyboard_name": "CANARY60RGB",
+    "url": "",
+    "maintainer": "canary",
+    "width": 15,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/canary/canary60rgb/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/canary/canary60rgb/info.json (hunk 13 above)
Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/info.json:6]: https://beta.docs.qmk.fm/developing-qmk/qmk-reference/reference_info_json
Based on the link above, info.json should have "width" and "height" parameters.

File: keyboards/canary/canary60rgb/v1/config.h (hunk 12 above)
Comment (AnthonyNguyen168) [reply keyboards/canary/canary60rgb/v1/config.h]: Done.

**Comment by AnthonyNguyen168** [comment]
I committed the necessary changes and answer the questions.
Please consider this pull request.

File: keyboards/canary/canary60rgb/info.json (hunk 13 above)
Comment (fauxpark) [reply keyboards/canary/canary60rgb/info.json:6]: https://github.com/qmk/qmk_firmware/pull/14274

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by AnthonyNguyen168** [comment]
I did commit with your request.
Please check again.

**Review by drashna — APPROVED** [review]

**Comment by AnthonyNguyen168** [comment]
Dear QMK maintainers,
Please check this pull request. :)
Thanks.

**Comment by AnthonyNguyen168** [comment]
Dear @fauxpark and @zvecr,
Please check this pull request. :)
Many thanks.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**